sway/input/cursor.c

```diff
@@ -162,33 +170,23 @@ void cursor_send_pointer_motion(struct sway_cursor *cursor, uint32_t time_msec,
 				seat_set_focus_warp(cursor->seat, c, false);
 			}
 		} else if (c->type == C_VIEW) {
-			// Don't switch focus on title mouseover for
-			// stacked and tabbed layouts
-			// If pointed container is in nested containers which are
-			// inside tabbed/stacked layout we should skip them
-			bool do_mouse_focus = true;
-			bool is_visible = view_is_visible(c->sway_view);
-			struct sway_container *p = c->parent;
-			while (p) {
-				if ((p->layout == L_TABBED || p->layout == L_STACKED)
-					&& !is_visible) {
-					do_mouse_focus = false;
-					break;
-				}
-				p = p->parent;
-			}
-			if (!do_mouse_focus) {
-				struct sway_container *next_focus = seat_get_focus_inactive(
-						cursor->seat, p);
-				if(next_focus && !sway_assert(next_focus->type == C_VIEW,
-							"focus inactive container is not a view")) {
-					return;
-				}
-				if (next_focus && view_is_visible(next_focus->sway_view)) {
-					seat_set_focus_warp(cursor->seat, next_focus, false);
-				}
-			} else {
+			// Focus c if both of the following are true:
+			// - cursor is over a new view, i.e. entered a new window; and
+			// - the new view is visible, i.e. not hidden in a stack or tab.
+			if (c != prev_c && view_is_visible(c->sway_view)) {
 				seat_set_focus_warp(cursor->seat, c, false);
+			} else {
+				struct sway_container *next_focus =
+					seat_get_focus_inactive(cursor->seat, &root_container);
+				if (next_focus) {
+					if (!sway_assert(next_focus->type == C_VIEW,
```

Comment:
This can't be an assert, `seat_get_focus_inactive` can return a workspace

Reply:
Okay, I changed this to just check the type.